### PR TITLE
[DEVOPS-453] Development topology fixes

### PR DESCRIPTION
**Pull request by deepfire**

### CODE REVIEW

File: globals.nix [hunk 1]
```diff
@@ -74,7 +74,7 @@ let topologySpec     = builtins.fromJSON (builtins.readFile topologyFile);
     relays          = filter     (x: x.value.typeIsRelay)             indexed;
     nodeMap         = listToAttrs (cores ++ relays);
     # WARNING: this depends on the sort order, as explained above.
-    firstRelay      = findFirst (x: x.value.typeIsRelay) (throw "No relays in topology, it's sad we can't live, but then.. who does?") indexed;
+    firstRelay      = findFirst (x: x.value.typeIsRelay) ({ name = "fake-non-relay"; value = { type = "relay"; i = builtins.length indexed; }; }) indexed;
```
Comment (domenkozar) [inline globals.nix]: why is suddently there a need for fake relay where previously it was an evaluation error?

File: default.nix [hunk 2]
```diff
@@ -38,10 +38,10 @@ in {
              (compiler.callPackage ./iohk/default.nix {})
              (drv: {
                 executableToolDepends = [ pkgs.makeWrapper ];
-                libraryHaskellDepends = iohk-ops-extra-runtime-deps;
+                libraryHaskellDepends = iohk-ops-extra-runtime-deps ++ [ cardano-sl-pkgs.cardano-sl-auxx pkgs.file nixops ];
```
Comment (domenkozar) [inline default.nix]: Why are these needed to be in nix-shell?

Comment (deepfire) [reply default.nix]: auxx and file is for the update proposals
keygen is for dev cluster deployments

File: globals.nix (hunk 1 above)
Comment (deepfire) [reply globals.nix]: we can evaluate without nodes now

Comment (domenkozar) [reply globals.nix]: But shouldn't we skip the whole topology evaluation if there are no nodes?

Comment (deepfire) [reply globals.nix]: Ideally, yes.

**Comment by deepfire** [comment]
All issue ids have been added.

File: default.nix [hunk 3]
```diff
@@ -38,10 +41,10 @@ in {
              (compiler.callPackage ./iohk/default.nix {})
              (drv: {
                 executableToolDepends = [ pkgs.makeWrapper ];
-                libraryHaskellDepends = iohk-ops-extra-runtime-deps;
+                libraryHaskellDepends = iohk-ops-extra-runtime-deps ++ [ nixops ];
```
Comment (domenkozar) [inline default.nix]: shouldn't nixops be part of `iohk-ops-extra-runtime-deps` as well?

Comment (deepfire) [reply default.nix]: Done..

**Comment by deepfire** [comment]
@domenkozar, the nodeless-ness PR was separated into https://github.com/input-output-hk/iohk-ops/pull/182

File: default.nix [hunk 4]
```diff
@@ -34,6 +23,23 @@ in {
         sha256 = "1fly6ry7ksj7v5rl27jg5mnxdbjwn40kk47gplyvslpvijk65m4q";
       };
     in (import "${nixopsUnstable}/release.nix" {}).build.${system};
+  iohk-ops-extra-runtime-deps = [
+    pkgs.gitFull pkgs.nix-prefetch-scripts compiler.yaml
```
Comment (domenkozar) [inline default.nix]: Do we really need gitFull?

Comment (domenkozar) [reply default.nix]: It only adds 50MB, so looks OKish.

File: shell.nix [hunk 5]
```diff
@@ -11,7 +11,7 @@ let
 in pkgs.lib.overrideDerivation iohk-ops.env
    (old: {
      shellHook = ''
-       io () {
+       function io {
```
Comment (domenkozar) [inline shell.nix]: @deepfire what's the reason for this change? If I understand correctly `io ()` is POSIX.

Comment (cleverca22) [reply shell.nix]: i also ran into this failing on the deployer, couldnt figure out why it was a problem

**Comment by deepfire** [comment]
I can add an arbitrary commit id -- it's really a generic cosmetic thing, not attached anywhere and not meriting an issue of its own.

File: shell.nix (hunk 5 above)
Comment (domenkozar) [reply shell.nix]: Can you paste the error?

Comment (domenkozar) [reply shell.nix]: Aha was it:

```
$ nix-shell
bash: eval: line 69: syntax error near unexpected token `('
bash: eval: line 69: `io () {'
``

Comment (cleverca22) [reply shell.nix]: yep, exactly

Comment (domenkozar) [reply shell.nix]: Really strange, bash should support this syntax.
